src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala

```diff
@@ -51,11 +54,19 @@ class DagInstanceRepositoryImpl extends DagInstanceRepository {
     } yield ()).transactionally
   ).map(_ => (): Unit)
 
-  override def getNewActiveDags(idToFilter: Seq[Long], size: Int): Future[Seq[DagInstance]] = db.run(
-    dagInstanceTable.filter(ji =>
-      !ji.id.inSet(idToFilter) && ji.status.inSet(DagInstanceStatuses.finalStatuses)
-    ).take(size).result
-  )
+  def getDagsToRun(idToFilter: Seq[Long], size: Int)(implicit executionContext: ExecutionContext): Future[Seq[DagInstance]] = {
+    val prefiltredResult = db.run(
+      dagInstanceTable.filter { di =>
+        di.workflowId.in(
+          dagInstanceTable.filter(_.id.inSet(idToFilter)).map(_.workflowId)
+        ) || di.status.inSet(DagInstanceStatuses.finalStatuses)
```

Comment:
Am I reading this wrong or will this run things that have status final?

Reply:
Changed. Try now.